x-pack/solutions/observability/plugins/observability_ai_assistant_app/server/functions/query/validate_esql_query.ts

```diff
@@ -16,9 +16,11 @@ import { splitIntoCommands } from '@kbn/inference-plugin/common';
 export async function runAndValidateEsqlQuery({
   query,
   client,
+  signal,
 }: {
   query: string;
   client: ElasticsearchClient;
+  signal: AbortSignal;
```

Comment:
Added missing `signal`. This should mean that we cancel the ES request when appropriate.